lua/astrocommunity/pack/hyprland/init.lua

```diff
@@ -0,0 +1,65 @@
+---@type LazySpec
+return {
+  -- mason ensure install hyprls
+  {
+    "WhoIsSethDaniel/mason-tool-installer.nvim",
+    optional = true,
+    opts = function(_, opts)
+      opts.ensure_installed = require("astrocore").list_insert_unique(opts.ensure_installed, { "hyprls" })
+    end,
+  },
+
+  {
+    "AstroNvim/astrocore",
+    optional = true,
+    ---@type AstroCoreOpts
+    opts = {
+      -- add "hyprlang" file tpye
+      filetypes = {
+        extension = {
+          hl = "hyprlang",
+        },
+        filename = {
+          ["hyprland.conf"] = "hyprlang",
+        },
+        pattern = {
+          ["*/hypr/*.conf"] = "hyprlang",
+          ["hypr*.conf"] = "hyprlang",
+        },
+      },
+
+      -- hyprls client auto-attach by filetype
+      autocmds = {
+        hyprland_support = {
+          {
+            desc = "Hyprls auto-attach",
+            event = { "FileType" },
+            pattern = { "hyprlang" },
+            callback = function()
+              local core = require "astrocore"
+              if vim.fn.executable "hyprls" ~= 0 then
+                vim.lsp.start { name = "hyprls", cmd = { "hyprls" }, root_dir = vim.fn.getcwd() }
+              else
+                core.notify(
+                  "`hyprls` not installed yet. Install it by Mason or running `go install hyprls` to enable hyprland language support!",
+                  vim.log.levels.WARN
+                )
+              end
+            end,
+          },
+        },
+      },
+    },
+  },
+
```

Comment:
```suggestion
  {
    "AstroNvim/astrolsp",
    opts = function(_, opts)
      local astrocore = require "astrocore"
      opts.servers = astrocore.list_insert_unique(opts.servers, { "hyprls" })

      return astrocore.extend_tbl(opts, {
        config = {
          hyprls = { filetypes = { "hyprlang" } },
        },
      })
    end,
  },
```

This tells AstroLSP to configure the server in `lspconfig` since it's not currently supported in `mason-lspconfig` as well as configure the filetypes correctly. Once that is merged in, we can simplify this to:

```lua
  {
    "AstroNvim/astrolsp",
    opts = {
      config = {
        hyprls = { filetypes = { "hyprlang" } },
      },
    },
  },
```